conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/BodySerDe.java

```diff
@@ -19,15 +19,34 @@
 import io.undertow.server.HttpServerExchange;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Optional;
 
 /** Request and response Deserialization and Serialization functionality used by generated code. */
 public interface BodySerDe {
 
     /** Creates a {@link Serializer} for the requested type. Serializer instances should be reused. */
-    <T> Serializer<T> serializer(TypeMarker<T> type);
+    default <T> Serializer<T> serializer(TypeMarker<T> type, Endpoint endpoint) {
+        return serializer(type, Optional.of(endpoint));
+    }
+
+    /** Kept for backwards compatibility. */
+    default <T> Serializer<T> serializer(TypeMarker<T> type) {
+        return serializer(type, Optional.empty());
+    }
+
+    <T> Serializer<T> serializer(TypeMarker<T> type, Optional<Endpoint> endpoint);
```

Comment:
Let's minimize the API surface area by only adding a new `default <T> Serializer<T> serializer(TypeMarker<T> type, Endpoint endpoint) {` which delegates to the existing method, but without the endpoint arg.

As is this change would be an abi break for any consumer who has implemented their own BodySerDe (which they aren't supposed to do, but I'm aware of several)

Reply:
Makes sense, updated.